src/commonMain/kotlin/arrow/exact/ExactDsl.kt

```diff
@@ -0,0 +1,37 @@
+package arrow.exact
+
+import arrow.core.Either
+import arrow.core.raise.Raise
+import arrow.core.raise.either
+
+@DslMarker public annotation class ExactDsl
+
+@ExactDsl
+public fun <A, R> exact(construct: ExactScope<A, ExactError>.() -> R): Exact<A, R> =
+  object : Exact<A, R> {
+    override fun from(value: A): Either<ExactError, R> = either { construct(ExactScope(value, this)) }
+  }
+
+@ExactDsl
+public fun <E : Any, A, R> exactEither(construct: ExactScope<A, E>.() -> R): ExactEither<E, A, R> =
+  object : ExactEither<E, A, R> {
+    override fun from(value: A): Either<E, R> = either { construct(ExactScope(value, this)) }
+  }
+
+
+public class ExactScope<A, E : Any>(public val raw: A, private val raise: Raise<E>) : Raise<E> by raise {
+
+  public fun <B> ensure(exact: ExactEither<E, A, B>): B {
```

Comment:
Can you annotate it the `ExactDsl` dsl marker so Android Studio and IntelliJ can highlight it?